dashboard/cron.py

```diff
@@ -615,7 +615,10 @@ def do(self):
             return (status,)
 
         # Lock in valid course IDs that data will be pulled for.
-        self.valid_locked_course_ids = course_verification.course_data['id'].to_list()
+        if not course_verification.course_data.empty:
```

Comment:
This would also be the same as
course_verification.course_data.get("id", pd.Series(dtype=pd.Int64Dtype)).to_list()

Reply:
Shorter, but probably less readable?

Reply:
This is mentioned as an [anti-pattern:](https://docs.quantifiedcode.com/python-anti-patterns/correctness/not_using_get_to_return_a_default_value_from_a_dictionary.html)

"Although there is nothing wrong with this code, it is verbose and inefficient because it queries the dictionary twice." 

I think it's only less readable because we have to pass a pd.Series (and that in turn has a warning without a dtype on an empty Series) but that also makes it more explicit too.

So it's fine either way. the performance isn't a big deal

Reply:
You're extrapolating a bit on that anti-pattern, but you are probably correct. I will make the change.

Reply:
It might be better also if `verify_course_ids` on line 146 returns this as a DataFrame

`pd.DataFrame(columns=["id", "canvas_id", "enrollment_term_id", "name", "start_at", "conclude_at"])`

Then you don't have to do either of these!

Reply:
Hm, I like that. Possibly more verbose, but I like that the `DataFrame` maintains the same structure.